### PR TITLE
Tidy and fix validation of Tagging / Metadata operations

### PR DESCRIPTION
**Pull request by TimCsaky**
# Description

ticket: https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-3248

enforce tag limit of 10 in Joi middleware
make tags/meta optional when adding meta/tags
tagset and metadata validation options can be given with function arguments but this is kind of redundant now.
remove temp validation from controllers
Remove obsolete unit tests

- thoroughly regression tested using postman

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by jujaga — DISMISSED** [review]
The TODO to eventually fold this into the validation logic needs to remain. Even if we don't do it now, it needs to eventually be moved there.

File: app/src/validators/common.js [hunk 1]
```diff
@@ -45,15 +45,27 @@ const type = {
...
   }),
 
-  metadata: (minKeyCount = 0, minValueStringLength = 0) => Joi.object()
+  /**
+   * custom metadata (object) type schema with parameters
+   * @param minKeyCount minimum number of tags allowed
+   * @param minValueStringLength minimum string length of tag value allowed
+   * @param maxKeyCount maximum number of tags allowed (default ot 9 because COMS also adds a `coms-id` tag by default)
+   */
+  metadata: ({ minKeyCount = 0, minValueStringLength = 1 } = {}) => Joi.object()
```
Comment (TimCsaky) [inline app/src/validators/common.js:54]: now that we are making meta params optional for addMetadata and replaceMetadata operations, these arguments are kind of redundant. 

same for tagset().. but they could be useful some time in the future. oh well

File: app/src/controllers/object.js [hunk 2]
```diff
@@ -176,47 +168,48 @@ const controller = {
...
+        // remove existing 'coms-id' tag if it exists
+        .filter(x => x.Key !== 'coms-id')
+        // filter duplicates
+        .filter((element, idx, arr) => arr.findIndex(element2 => (element2.Key === element.Key)) === idx)
+        // add 'coms-id' tag
+        .concat([{ Key: 'coms-id', Value: objId }]);
+
+      if (newSet.length > 10) {
+        // 409 when total tag limit exceeded
+        return new Problem(409, { detail: 'User-defined Tag count limit is 9' }).send(res);
```
Comment (jujaga) [inline app/src/controllers/object.js:192]: `Request exceeds maximum of 9 user-defined tag sets allowed`

File: app/src/validators/common.js [hunk 3]
```diff
@@ -45,15 +45,27 @@ const type = {
...
     .pattern(/^x-amz-meta-.{1,255}$/i, Joi.string().min(minValueStringLength).max(255), { matches: Joi.array().min(minKeyCount) })
     .unknown(),
 
-  tagset: (minKeyCount = 1, minValueStringLength = 0) => Joi.object()
+  /**
+   * custom tagset (object) type schema with parameters
+   * @param minKeyCount minimum number of tags allowed
+   * @param minValueStringLength minimum string length of tag value allowed
+   * @param maxKeyCount maximum number of tags allowed (default ot 9 because COMS also adds a `coms-id` tag by default)
```
Comment (jujaga) [inline app/src/validators/common.js:62]: JSDocs do not match function signature. Params should be marked as optional (brackets) with a default value specified. Should also document `@returns object Joi object` or similar.

**Review by jujaga — APPROVED** [review]

**Review by kamorel — APPROVED** [review]

**Pull request merged**